packages/foam-vscode/src/dated-notes.ts

```diff
@@ -1,10 +1,83 @@
-import { workspace, WorkspaceConfiguration } from 'vscode';
+import {
+  QuickPickItem,
+  window,
+  workspace,
+  WorkspaceConfiguration,
+} from 'vscode';
 import dateFormat from 'dateformat';
 import { existsInFs } from './core/utils/path';
 import { focusNote } from './utils';
 import { URI } from './core/model/uri';
 import { fromVsCodeUri } from './utils/vsc-utils';
 import { NoteFactory } from './services/templates';
+import { setFlagsFromString } from 'v8';
+
+export async function openDailyNoteForToday() {
+  console.log('Open daily note for today');
+  openDailyNoteFor();
+}
+
+class DateItem implements QuickPickItem {
+  label: string;
+  date: Date;
+
+  constructor(public l: string, public d: Date) {
+    this.label = l;
+    this.date = d;
+  }
+}
+
+function generateDateItems(): DateItem[] {
+  const items = [];
+
+  // Past
+  for (let offset = -5; offset < -1; offset++) {
+    const d = new Date();
+    d.setDate(d.getDate() + offset);
+    items.push(new DateItem(d.toDateString(), d));
+  }
+
+  // Yesterday
+  const yesterday = new Date();
+  yesterday.setDate(yesterday.getDate() - 1);
+  items.push(new DateItem('yesterday', yesterday));
+
+  // Today
+  const today = new DateItem('today', new Date());
+  items.push(today);
+
+  // Tomorrow
+  const tomorrow = new Date();
+  tomorrow.setDate(tomorrow.getDate() + 1);
+  items.push(new DateItem('tomorrow', tomorrow));
+
+  // Future
+  for (let offset = 2; offset < 6; offset++) {
+    const d = new Date();
+    d.setDate(d.getDate() + offset);
+    items.push(new DateItem(d.toDateString(), d));
+  }
+
+  return items;
+}
+
+export async function openDailyNoteForPickedDate() {
+  console.log('Open daily note for date');
+
+  const result = await window.showQuickPick<DateItem>(generateDateItems(), {
```

Comment:
I believe you should be able to have a default by looking at the API, am I missing something? https://code.visualstudio.com/api/references/vscode-api#QuickPickItem

Reply:
That's what I thought but the `picked` property is about items that are selected if you allow `canPickMany`.

Reply:
yup, you are right, I got confused :)